src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,27 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        List<PackageReference> installedProjectReferences = new List<PackageReference>(projectReferences.Count());
+
+                        // Since ProjectReferences are not in the dependencies section of the package spec (installed packages), we need to manually add them
+                        // to the installed packages list to compute transitive origins
+                        foreach (var projectReference in projectReferences)
+                        {
+                            var packageReference = new PackageReference(
+                                identity: new PackageIdentity(projectReference.Name, projectReference.Version),
+                                targetFramework: new NuGetFramework(projectReference.Framework),
+                                userInstalled: false,
+                                developmentDependency: false,
+                                requireReinstallation: false,
+                                allowedVersions: VersionRange.Parse(projectReference.Version.ToString()));
```

Comment:
Isn't that a version object? Can we just create a version range with the version object as a lower bound?